docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md

```diff
@@ -0,0 +1,103 @@
+1. Querying the content API, or via some other means, to determine the list of interactive URLs to press
+2. For every URL make a request to /interactive-librarian/live-presser/{path}
+3. For every successfully pressed article make a request to /interactive-librarian/read-clean-write/{path}
+
+To press a single interactive we can use the frontend admin tool. On the admin tool, select the new option ‘Press an interactive’, enter the full URL for the interactive, click ‘Press’ and wait for the response. If there’s an error in the response it’ll need to be reported to the dotcom team.
+
+**How do we know a page is pressed?**
+
+Each pressed article will have additional copy appended to the document. At the very bottom of the page we should see the copy ‘This article was archived on <date>. Some elements may be out of date.’
+
+**How do we decide to serve pressed content to readers?**
+
+We have an intermediate solution for determining if a page is pressed: we retain a list of article URLs in config. Only if the article being requested is included in this list do we serve the pressed content.
+
+In the long term we’d like to mark pressed articles with a tracking tag (tracking/dcroptout). Articles with this tag will be served as pressed versions. We can’t adopt this solution until Ed Tools have completed a spike to understand how we can achieve this.
+
+**How can we view a pressed page?**
+To view a pressed page there are a couple of options:
+- Get the document from S3 directly (aws-frontend-archive).
+- Intermediate solution: add the interactive path to the frontend config (https://github.com/guardian/frontend/blob/dlawes/serve-pressed-interactives/common/app/services/dotcomrendering/PressedInteractives.scala#L11).
+- Long-term solution: add tag tracking/dcroptout to article.
+
+**Can we opt-out of pressing and render via frontend or DCR?**
+
+Yes! In the short term, if we want to opt-out of serving pressed content then we omit or remove the interactive path in the relevant frontend config. In the long term, if we want to opt-out of serving the pressed content we will remove, or we will not apply, the chosen tracking tag (tracking/dcroptout).
+
+Note that, after some agreed period of time, I believe our intention would be to remove the rendering code from the frontend codebase, after which point articles could only be rendered via DCR.
+
+**What about interactives via AMP?**
+
+Interactives have to opt-in to rendering via AMP using a specific tag. Interactives rendering via AMP are out of scope for this migration.
+
+About rendering interactives for apps, I believe it’s a separate flow. Given we’re not changing the article content then legacy interactives for mobile will be unaffected. We need to be aware that if there is a request to update the content of a legacy interactive this would need to be done via Composer for the mobile version (so the CAPI response is updated), as well as updating the pressed version for web.
```

Comment:
Worth adding a heading here for Apps? So as to distinguish from AMP.

Reply:
Yes, will do!